src/h5web/App.tsx

```diff
@@ -1,33 +1,33 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import { ReflexContainer, ReflexSplitter, ReflexElement } from 'react-reflex';
 
 import Explorer from './explorer/Explorer';
 import DatasetVisualizer from './dataset-visualizer/DatasetVisualizer';
-import { HDF5Link, HDF5Collection, HDF5HardLink } from './providers/models';
+import { HDF5Link, HDF5Collection, HDF5Entity } from './providers/models';
 import MetadataViewer from './metadata-viewer/MetadataViewer';
 import styles from './App.module.css';
-import { isHardLink } from './providers/type-guards';
+import { isDataset } from './providers/type-guards';
+import { useEntity } from './providers/hooks';
 
 function App(): JSX.Element {
   const [selectedLink, setSelectedLink] = useState<HDF5Link>();
-  const [selectedDataset, setSelectedDataset] = useState<HDF5HardLink>();
+  const [selectedDataset, setSelectedDataset] = useState<
+    HDF5Entity<HDF5Collection.Datasets>
+  >();
+
+  const selectedEntity = useEntity(selectedLink);
+
+  useEffect(() => {
+    if (selectedEntity && isDataset(selectedEntity)) {
+      setSelectedDataset(selectedEntity);
+    }
+  }, [selectedEntity]);
 
   return (
     <div className={styles.app}>
       <ReflexContainer orientation="vertical" windowResizeAware>
         <ReflexElement className={styles.explorer} flex={0.3} minSize={250}>
-          <Explorer
-            onSelect={link => {
-              setSelectedLink(link);
-
-              if (
-                isHardLink(link) &&
```

Comment:
The hard link check is now performed inside `useEntity` => it returns undefined if the link is not a hard link, which is then checked inside `useEffect`.